drivers/gpio/gpio_gecko.c

```diff
@@ -286,7 +286,7 @@ DEVICE_DT_DEFINE(DT_INST(0, silabs_gecko_gpio),
 		    gpio_gecko_common_init,
 		    NULL,
 		    &gpio_gecko_common_data, &gpio_gecko_common_config,
-		    POST_KERNEL, CONFIG_GPIO_GECKO_COMMON_INIT_PRIORITY,
+		    POST_KERNEL, CONFIG_GPIO_INIT_PRIORITY,
```

Comment:
After further tests: This does not work. The init priority used here must be lower than the init priority used in line 328.
Reason: In the function gpio_gecko_common_init() initializes the gpio peripheral that contains all the ports (name "GPIO"). In line 328, the individual ports (e.g. "GPIO_A") are initialized. The "GPIO" peripheral must be initialized before the individual ports.

With the current code the device test  tests/drivers/gpio/gpio_api_1pin fails. When using the value 39 here, it works again.

Reply:
Thanks for testing. I pushed a new version that sets the value to 39 for gecko.

Reply:
Thanks for the change, sorry for being unclear. The test still fails.

It does not work when the init priority here (line 289) is the same as in line 328. The init priority here must be below the one in line 328. Before your code change in this line 39 was used and in line 328 40 was used.

I think we still need the additional define for this line.

Reply:
I misunderstood your earlier comment, thanks for clarifying. I pushed another version and added a build error to catch if the common init is after the individual port init, please have a look.